feature/metis/conversation/src/test/kotlin/de/tum/informatics/www1/artemis/native_app/feature/metis/conversation/ui/ConversationProfilePictureE2eTest.kt

```diff
@@ -0,0 +1,127 @@
+
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.test.junit4.ComposeTestRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.unit.dp
+import de.tum.informatics.www1.artemis.native_app.core.common.test.UnitTest
+import de.tum.informatics.www1.artemis.native_app.core.test.BaseComposeTest
+import de.tum.informatics.www1.artemis.native_app.core.test.test_setup.DefaultTimeoutMillis
+import de.tum.informatics.www1.artemis.native_app.feature.metis.conversation.TestInitialReplyTextProvider
+import de.tum.informatics.www1.artemis.native_app.feature.metis.conversation.service.impl.EmojiServiceStub
+import de.tum.informatics.www1.artemis.native_app.feature.metis.conversation.ui.chatlist.ChatListItem
+import de.tum.informatics.www1.artemis.native_app.feature.metis.conversation.ui.chatlist.MetisChatList
+import de.tum.informatics.www1.artemis.native_app.feature.metis.conversation.ui.chatlist.PostsDataState
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.content.dto.UserRole
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.db.pojo.PostPojo
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.ui.profile_picture.TEST_TAG_PROFILE_PICTURE_IMAGE
+import de.tum.informatics.www1.artemis.native_app.feature.metis.shared.ui.profile_picture.TEST_TAG_PROFILE_PICTURE_INITIALS
+import de.tum.informatics.www1.artemis.native_app.feature.metistest.ProfilePictureImageProviderStub
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.test.runTest
+import kotlinx.datetime.Clock
+import org.junit.Test
+import org.junit.experimental.categories.Category
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@Category(UnitTest::class)
+@RunWith(RobolectricTestRunner::class)
+class ConversationProfilePictureUiTest : BaseComposeTest() {
+
+    private val clientId = 20L
+    private val courseId = 1L
+
+    private var profilePictureImageProvider = ProfilePictureImageProviderStub()
+
+    private val postWithAuthorImageUrl = PostPojo(
+        clientPostId = "client-id",
+        serverPostId = 12,
+        content = "Post content",
+        resolved = false,
+        updatedDate = null,
+        creationDate = Clock.System.now(),
+        authorId = clientId,
+        title = null,
+        authorName = "author name",
+        authorRole = UserRole.USER,
+        authorImageUrl = "authorImageUrl",
+        courseWideContext = null,
+        tags = emptyList(),
+        answers = emptyList(),
+        reactions = emptyList()
+    )
+
+    @Test
+    fun `test GIVEN a post with an authorImageUrl WHEN displaying the post THEN the profile picture is shown`() = runTest {
+        val post = postWithAuthorImageUrl
+        setupUi(post)
+
+        composeTestRule.waitUntilNoAssertionError {
```

Comment:
Is this really necessary? This could delay our tests by seconds.

I think it should be possible to rewrite this test to not use a waitUntil.

Reply:
Unfortunately I think it is necessary, because we have to wait for the AsyncImageLoader state to switch to Success, and I found no way of enforcing this in tests. We can also not inherit from AsyncImageLoader and override the state. If you know a way to circumvent this, I am more than happy to implement this

Reply:
UPDATE: in #172 I found a way where we do not need this anymore (with coil3)